src/jenkins_job_insight/storage.py

```
@@ -478,6 +676,793 @@ def _count_child_failures_recursive(child: dict) -> int:
+            If empty, the DB default (CURRENT_TIMESTAMP) is used.
+    """
+    logger.debug(f"populate_failure_history: job_id={job_id}")
+    job_name = result_data.get("job_name", "")
+    build_number = result_data.get("build_number", 0)
+
+    rows = _extract_failures_for_history(
+        result_data, job_id, job_name, build_number, analyzed_at=analyzed_at
+    )
+    if not rows:
+        logger.debug(
+            f"populate_failure_history: job_id={job_id}, no failures to insert"
```

Ruff (0.15.6)

[warning] Logging statement uses f-string

(G004)